### ISSUE TITLE
make `--cert-identity` a required parameter in `sigstore verify`

### ISSUE THREAD

**Issue by graingert** [sigstore/sigstore-python#155]
**Description**

Forgetting to verify who a certificate is actually from is a foot gun, which accidentally caught out urllib3 see https://github.com/urllib3/urllib3/issues/2675

**Comment by woodruffw** [comment]
This is also partially blocked on #108, since `--cert-identity` is pretty limited in scope.

**Comment by haydentherapper** [comment]
We are likely going to add this very soon in Cosign. Will tag y'all on the PR for discussion.

**Comment by woodruffw** [comment]
#299 currently contains this change on our side, although I may break it out into a separate PR.

**Comment by woodruffw** [comment]
This is done.
